Sources/PackageCollections/Providers/GitHubPackageMetadataProvider.swift

```diff
@@ -62,28 +65,52 @@ struct GitHubPackageMetadataProvider: PackageMetadataProvider {
 
             // get the main data
             sync.enter()
-            let options = self.makeRequestOptions(validResponseCodes: [200])
-            var headers = HTTPClientHeaders()
-            headers.add(name: "Accept", value: "application/vnd.github.mercy-preview+json")
-            httpClient.get(metadataURL, headers: headers, options: options) { result in
+            var metadataHeaders = self.makeRequestHeaders(metadataURL)
+            metadataHeaders.add(name: "Accept", value: "application/vnd.github.mercy-preview+json")
+            let metadataOptions = self.makeRequestOptions(validResponseCodes: [200, 401, 403, 404])
+            httpClient.get(metadataURL, headers: metadataHeaders, options: metadataOptions) { result in
                 defer { sync.leave() }
                 resultsLock.withLock {
                     results[metadataURL] = result
                 }
-                // if successful, fan out multiple API calls
-                if case .success = result {
-                    [tagsURL, contributorsURL, readmeURL].forEach { url in
-                        sync.enter()
-                        httpClient.get(url, options: options) { result in
-                            defer { sync.leave() }
-                            resultsLock.withLock {
-                                results[url] = result
+                if case .success(let response) = result {
+                    let apiLimit = response.headers.get("X-RateLimit-Limit").first.flatMap(Int.init) ?? -1
+                    let apiRemaining = response.headers.get("X-RateLimit-Remaining").first.flatMap(Int.init) ?? -1
+
+                    switch (response.statusCode, metadataHeaders.contains("Authorization"), apiRemaining) {
+                    case (_, _, 0):
+                        self.diagnosticsEngine?.emit(warning: "Exceeded API limits on \(metadataURL.host ?? metadataURL.absoluteString) (\(apiRemaining)/\(apiLimit)), consider configuring a API token for this service.")
```

Comment:
> consider configuring a API token for this service

consider configuring *an* API token for this service

(GitHub doesn't seem to let me quote/suggest edits anymore?!)